planner/core/logical_plan_builder.go

```diff
@@ -3841,6 +3842,26 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		// Try CTE.
 		p, err := b.tryBuildCTE(ctx, tn, asName)
 		if err != nil || p != nil {
+			if len(b.cteNameProcessingStack) > 0 {
+				if tblIds, ok := b.cteToTblIDMapper[tn.Name.L]; ok {
+					cteName := b.cteNameProcessingStack[len(b.cteNameProcessingStack)-1]
+					b.cteToTblIDMapper[cteName] = append(b.cteToTblIDMapper[cteName], tblIds...)
+				}
+			} else {
+				if tblIds, ok := b.cteToTblIDMapper[tn.Name.L]; ok {
+					for _, tblID := range tblIds {
+						if _, ok := b.selTmpTblIds[tblID]; ok {
+							tb, exist := b.is.TableByID(tblID)
+							if !exist {
+								return nil, infoschema.ErrTableNotExists.GenWithStack("Table which ID = %d does not exist.", tblID)
+							}
+							errMsg := fmt.Sprintf(errno.MySQLErrName[errno.ErrCantReopenTable].Raw, tb.Meta().Name.L)
+							return nil, ErrInternal.GenWithStack(errMsg)
```

Comment:
how about `return nil, ErrInternal.GenWithStack(errno.MySQLErrName[errno.ErrCantReopenTable].Raw,tb.Meta().Name.L)`

Reply:
done

Reply:
The correct usage should be:

1. define a global error object in this package, maybe error.go

```
errCantReopenTable =  dbterror.ClassTypes.NewStdErr(errno.ErrCantReopenTable, mysql.MySQLErrName[errno.ErrCantReopenTable])
```

2. and then use  `errCantReopenTable.GenWithStackByArgs(tblName)` here


In this way, you can get the correct error code 1137, instead of 1815

@zhaoxugang